server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileProcessor.java

```diff
@@ -366,6 +415,87 @@ private void checkMemCostAndAddToTspInfo(InsertTabletPlan insertTabletPlan, int
     updateMemoryInfo(memTableIncrement, chunkMetadataIncrement, textDataIncrement);
   }
 
+  private void checkMemCostAndAddToTspInfo(InsertSinglePointPlan insertSinglePointPlan)
+      throws WriteProcessException {
+    // memory of increased PrimitiveArray and TEXT values, e.g., add a long[128], add 128*8
+    long memTableIncrement = 0L;
+    long textDataIncrement = 0L;
+    long chunkMetadataIncrement = 0L;
+    String deviceId = insertSinglePointPlan.getDeviceId().getFullPath();
+
+    if (workMemTable.checkIfChunkDoesNotExist(deviceId, insertSinglePointPlan.getMeasurement())) {
+      // ChunkMetadataIncrement
+      MeasurementMNode mnode = insertSinglePointPlan.getMeasurementMNode();
+      IMeasurementSchema schema = insertSinglePointPlan.getMeasurementMNode().getSchema();
+      if (schema.getType() == TSDataType.VECTOR) {
+        chunkMetadataIncrement +=
+            schema.getValueTSDataTypeList().size()
+                * ChunkMetadata.calculateRamSize(
+                    schema.getValueMeasurementIdList().get(0),
+                    schema.getValueTSDataTypeList().get(0));
+        memTableIncrement += TVList.vectorTvListArrayMemSize(schema.getValueTSDataTypeList());
+      } else {
+        chunkMetadataIncrement +=
+            ChunkMetadata.calculateRamSize(
+                insertSinglePointPlan.getMeasurement(), insertSinglePointPlan.getDataType());
+        memTableIncrement += TVList.tvListArrayMemSize(insertSinglePointPlan.getDataType());
+      }
+    } else {
+      // here currentChunkPointNum >= 1
+      int currentChunkPointNum =
+          workMemTable.getCurrentChunkPointNum(deviceId, insertSinglePointPlan.getMeasurement());
+      memTableIncrement +=
+          (currentChunkPointNum % PrimitiveArrayManager.ARRAY_SIZE) == 0
+              ? TVList.tvListArrayMemSize(insertSinglePointPlan.getDataType())
+              : 0;
+    }
+    // TEXT data mem size
+    if (insertSinglePointPlan.getDataType() == TSDataType.TEXT) {
+      textDataIncrement += MemUtils.getBinarySize((Binary) insertSinglePointPlan.getValue());
+    }
+    updateMemoryInfo(memTableIncrement, chunkMetadataIncrement, textDataIncrement);
+  }
+
+  private void checkMemcheckMemCostAndAddToTspInfoCostAndAddToTspInfo(
+      InsertTabletPlan insertTabletPlan, int start, int end) throws WriteProcessException {
```

Comment:
Remove this?

Reply:
Thank you for your reply，This function is independent of insert single point。I've removed it